reverse_engineering/queries/selectedTablesSubQuery/PartitionsSubQueryForRetrievingTheTablesSelectedByTheUser.js

```diff
@@ -0,0 +1,31 @@
+const { QueryForRetrievingTheTablesSelectedByTheUser } = require('./QueryForRetrievingTheTablesSelectedByTheUser');
+const { getProjectedPropertiesNames } = require('./getProjectedPropertiesNames');
+
+class PartitionsSubQueryForRetrievingTheTablesSelectedByTheUser extends QueryForRetrievingTheTablesSelectedByTheUser {
+	constructor({ schemaToTablesMap }) {
+		super();
+		this.schemaToTablesMap = schemaToTablesMap;
+	}
+
+	getQuery() {
+		const projection = {
+			'tbl.object_id': 'tableId',
+			'tbl.name': 'tableName',
+			'sch.name': 'schemaName',
+		};
+
+		const query = this.queryForRetrievingTheTablesSelectedByTheUser({
```

Comment:
**TIP**: always try to favour composition over inheritance.

In this case, inheriting from `QueryForRetrievingTheTablesSelectedByTheUser` has very little value.

Instead of BEING a `QueryForRetrievingTheTablesSelectedByTheUser`, it would be better for this class to HAVE a private instance of `QueryForRetrievingTheTablesSelectedByTheUser`. That way, there will be no public `this.queryForRetrievingTheTablesSelectedByTheUser()` for example. This is applicable to all sub-queries.